src/components/EmailGraphs.vue

```diff
@@ -1,116 +0,0 @@
-<template>
-  <div class="graphs">
-    <div class="graph">
-      <router-link class="header" to="/graphs/frommetogmail">
-        <h3>{{ fmtgLabel }}</h3>
-      </router-link>
-      <LineChart
-        :title="fmtgLabel"
-        :times="times"
-        :fromMeToGmail="fromMeToGmail"
-        :width="width"
-        :height="height"
-      />
-    </div>
-    <div class="graph">
-      <router-link class="header" to="/graphs/frommetonongmail">
-        <h3>{{ fmtngLabel }}</h3>
-      </router-link>
-      <LineChart
-        :title="fmtngLabel"
-        :times="times"
-        :fromMeToGmail="fromMeToNonGmail"
-        :width="width"
-        :height="height"
-      />
-    </div>
-    <div class="graph">
-      <router-link class="header" to="/graphs/tomefromgmail">
-        <h3>{{ tmfgLabel }}</h3> </router-link
-      ><LineChart
-        :title="tmfgLabel"
-        :times="times"
-        :fromMeToGmail="toMeFromGmail"
-        :width="width"
-        :height="height"
-      />
-    </div>
-    <div class="graph">
-      <router-link class="header" to="/graphs/tomefromnongmail">
-        <h3>{{ tmfngLabel }}</h3>
-      </router-link>
-      <LineChart
-        :title="tmfngLabel"
-        :times="times"
-        :fromMeToGmail="toMeFromNonGmail"
-        :width="width"
-        :height="height"
-      />
-    </div>
-  </div>
-</template>
-
-<script>
-import { ref } from 'vue'
-import { useStore } from 'vuex'
-import LineChart from './LineChart.vue'
-import { rangeScan } from '../fetchStats.js'
-
-export default {
-  name: 'EmailGraphs',
-  components: {
-    LineChart
-  },
-  async setup() {
-    // Setup and fetch
-    const store = useStore()
-    const start = Math.floor(
-      new Date(new Date().getTime() - 60 * 60 * 24 * 7 * 1000).getTime() / 1000
-    )
-    const end = Math.floor(Date.now() / 1000)
-    const gmailStats = await rangeScan('micahtyong@gmail.com', start, end)
-
-    // Parse and commit full stats
-    const fullStats = {
-      times: gmailStats.times,
-      fromMeToGmail: gmailStats.fromMeToGmail,
-      fromMeToNonGmail: gmailStats.fromMeToNonGmail,
-      toMeFromGmail: gmailStats.toMeFromGmail,
-      toMeFromNonGmail: gmailStats.toMeFromNonGmail
-    }
-    store.commit('setStats', fullStats)
-
-    // Return
-    return {
-      times: ref(gmailStats.times),
-      fromMeToGmail: ref(gmailStats.fromMeToGmail),
```

Comment:
We can sync up on zoom on how to restructure this. There a lot of ways to do something in Vue, so it's probably easier to walk through this example :)

Reply:
Yeah, I'd love to hear your thoughts on this as I was pretty lost. How might I translate from what I know from React?